spock-core/src/main/java/org/spockframework/mock/runtime/MockMakerRegistry.java

```diff
@@ -0,0 +1,154 @@
+package org.spockframework.mock.runtime;
+
+import org.spockframework.mock.CannotCreateMockException;
+import org.spockframework.mock.IMockObject;
+import org.spockframework.mock.ISpockMockObject;
+import org.spockframework.util.InternalSpockError;
+import org.spockframework.util.Nullable;
+import org.spockframework.util.ThreadSafe;
+import spock.mock.MockMakerId;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.ServiceLoader;
+import java.util.stream.Collectors;
+
+import static org.spockframework.mock.runtime.IMockMaker.IMockCreationSettings;
+import static org.spockframework.mock.runtime.IMockMaker.ITypeMockability;
+
+@ThreadSafe
+public final class MockMakerRegistry {
+
+  private static final String NL = "\n";
+
+  public static MockMakerRegistry INSTANCE = createFromServiceLoader();
```

Comment:
We should probably move away from static fields and try to use the `RunContext` to store an instance of this registry. Now that it is somewhat dynamic. This way we could also make use of the configuration script to configure the preferred mock maker.

Reply:
The issue is the `MockUtil`, which needs access to `MockMakerRegistry` from a static context.
Or is there a way to access the right `RunContext ` from a static method?

Reply:
You can call `RunContext.get()` to get the current one.

Reply:
But wouldn't I get a new `RunContext` for each `Thread`. Which would then create a new `MockMakerRegistry` for each `Thread`. This seams a bit exessive.